File: frontend/src/app/tickets/model/user-ticket.ts
import {Ticket} from "./ticket";
import {Validation} from "./validation";
import {TicketType} from "./ticket-type";

export interface UserTicket {
  code: number
  ticketName: string
  ticketType: TicketType
  discount: string
  ticketPrice: string
  purchaseTime: string
  isValidated: boolean
  ticket: Ticket
  validation: Validation
}

File: backend/src/main/java/com/example/backend/model/UserTicket.java
package com.example.backend.model;

import javax.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@SuperBuilder
@NoArgsConstructor
@Table(name = "user_ticket")
public class UserTicket {
    @Id
    @Column(name = "code", nullable = false)
    private String code;

    @ManyToOne
    @JoinColumn(name = "ticket_id")
    private Ticket ticket;

    @ManyToOne
    @JoinColumn(name = "discount_id")
    private Discount discount;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_login", nullable = false)
    private User user;

    @OneToOne(mappedBy = "userTicket")
    @JoinColumn(name = "validation_id")
    private Validation validation;

    @Column(name = "purchase_time", nullable = false)
    private LocalDateTime purchaseTime;
}